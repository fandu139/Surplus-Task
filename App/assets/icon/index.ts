export const ICON_SHOW_PASSWORD = 'Icon-EyeOpen';
export const ICON_HIDE_PASSWORD = 'Icon-EyeClose';
export const ICON_MENU_HOME = 'Icon-Menu-Home';
export const ICON_MENU_HOME_ACTIVE = 'Menu-Home-Active';
export const ICON_MENU_LIST_ORDER = 'Icon-Menu-List-Order';
export const ICON_MENU_LIST_ORDER_ACTIVE = 'Menu-Order-Active';
export const ICON_MENU_ACCOUNT = 'Icon-Menu-Account';
export const ICON_MENU_ACCOUNT_ACTIVE = 'Menu-Account-Active';
export const ICON_MENU_CREATE_ORDER = 'Icon-Menu-Create-Order';
export const ICON_MENU_HELP = 'Icon-Menu-Help';
export const ICON_MENU_HELP_ACTIVE = 'Menu-Help-Active';
export const ICON_BACK_ARROW = 'Back-Arrow';
export const ICON_NEXT_ARROW = 'Next-Arrow';
export const ICON_ACCOUNT = 'Icon-Account';
export const ICON_CHAT = 'Icon-Chat';
export const ICON_CHAT_INVERT = 'Icon-Chat-Invert';
export const ICON_DRIVER = 'Icon-Driver';
export const ICON_HELP = 'Icon-Help';
export const ICON_LOGOUT = 'Icon-Logout';
export const ICON_PASSWORD = 'Icon-Password';
export const ICON_PRIVACY_POLICY = 'Icon-Security';
export const ICON_ARROW_RIGHT = 'Icon-Chevron-Right';
export const ICON_ARROW_DOWN = 'Icon-Chevron-Down';
export const ICON_ARROW_UP = 'Icon-Chevron-Up';
export const ICON_PIN_LOCATION_START = 'Icon-LocationPin-Start';
export const ICON_PIN_LOCATION_END = 'Icon-LocationPin-Gray';
export const ICON_SEARCH = 'Icon-Search';
export const ICON_MAP = 'Icon-Map';
export const ICON_CONTACT_LIST = 'Icon-Contact';
export const ICON_CLOSE = 'Icon-Close';
export const ICON_PACKAGE = 'Icon-Package';
export const ICON_CHECK = 'Icon-Check';
export const ICON_SORT = 'Icon-Sort';
export const ICON_WHATSAPP = 'Icon-Whatsapp';
export const ICON_ARROW_DIRECTION = 'Icon-Arrow-Right-Direction';
export const ICON_CREATE_ORDER = 'Icon-Create-Order';
export const ICON_POINT = 'Icon-Point';
export const ICON_TELEPHONE = 'Icon-Phone';
export const ICON_WARNING = 'Icon-Warning';
export const ICON_COPY = 'Icon-Copy-Clipboard';
export const ICON_SEND_CHAT = 'Send-Chat';
export const ICON_EMOTICON = 'Emoticon';
export const ICON_PLUS = 'Icon-Plus';
export const ICON_TRASH = 'Icon-Trash';
export const ICON_SCAN = 'Icon-Scan';
export const ICON_FLASH_ON = 'Icon-FlashOn';
export const ICON_FLASH_OFF = 'Icon-FlashOff';
export const ICON_PENCIL = 'Icon-Pencil';
export const ICON_PENCIL_V2 = 'Icon-Pencil-V2';
export const ICON_INFO = 'Icon-Info';
export const ICON_DATE_PICKER = 'Icon-Date-Picker';
export const ICON_UPLOAD = 'Icon-Upload';
export const ICON_INFO_BLUE = 'Icon-Info-Blue';
export const ICON_FILE = 'Icon-File';
export const ICON_CROSS_MARK = 'Icon-Cross-Mark';
