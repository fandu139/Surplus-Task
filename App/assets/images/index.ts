/* eslint-disable global-require */
export const SURPLUS_LOGO = require('./surplus_logo.png');
export const FOOD_WASTE_MASKOT = require('./food-waste-maskot.png');
