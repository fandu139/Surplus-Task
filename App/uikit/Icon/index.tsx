import React from 'react';
import { createIconSetFromIcoMoon } from 'react-native-vector-icons';
import iconConfig from '../../assets/icon/Icon.json';
import { StyleProp, TextStyle, TouchableOpacity } from 'react-native';
import AppStyles from '../../theme/appStyles';

interface Props {
  name?: string;
  size: number | undefined;
  style?: StyleProp<TextStyle>;
  color?: string | undefined;
  onPress?: () => void;
  disabled?: boolean;
  testID?: string;
  accessibilityLabel?: string;
}

const ColectionIcon = createIconSetFromIcoMoon(iconConfig);

const Icon: React.FC<Props> = ({
  name,
  size,
  style,
  color,
  onPress,
  disabled,
  testID,
  accessibilityLabel,
}: Props) => {
  if (name) {
    return (
      <TouchableOpacity
        disabled={disabled || onPress === undefined}
        onPress={onPress}
        testID={testID}
        accessibilityLabel={accessibilityLabel || testID}
        hitSlop={AppStyles.HitSlopValue}
      >
        <ColectionIcon
          name={name}
          size={size}
          style={{ ...(style as Record<string, unknown>), color }}
        />
      </TouchableOpacity>
    );
  }
  return null;
};

Icon.defaultProps = {
  name: undefined,
  color: undefined,
  style: {},
  disabled: false,
  onPress: undefined,
  testID: 'button-icon',
  accessibilityLabel: undefined,
};

export default Icon;
