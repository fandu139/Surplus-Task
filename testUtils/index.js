import AuthContext from '../App/context/Auth';
import React from 'react';
import { act, fireEvent, render } from '@testing-library/react-native';

const customRender = (ui, options) => {
  const isAuthenticated = options?.isAuthenticated || false;
  const renderOptions = options?.rendererOptions;
  const authContext = options?.authContext;
  return render(
    <AuthContext.Provider value={{ isAuthenticated, ...authContext }}>{ui}</AuthContext.Provider>,
    renderOptions,
  );
};

// override render method
export { customRender as render };

export function wait(ms) {
  return act(() => new Promise((r) => setTimeout(r, ms)));
}

// Due to debounce on Button UI Kit, we need to add milisecond waiting timer
export function fireAsyncPress(element, ms = 100) {
  fireEvent.press(element);
  return wait(ms);
}
