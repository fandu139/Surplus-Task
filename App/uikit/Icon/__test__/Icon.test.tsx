import Icon from '../';
import React from 'react';
import { render } from '../../../../testUtils';
import { ICON_SHOW_PASSWORD } from '../../../assets/icon';

describe('Icon component test', () => {
  describe('Icon snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(<Icon name={ICON_SHOW_PASSWORD} size={16} color="black" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Icon without Name snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(<Icon size={16} color="black" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
