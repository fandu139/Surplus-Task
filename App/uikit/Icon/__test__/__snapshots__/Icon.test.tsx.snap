// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component test Icon snapshot test should render correctly 1`] = `
<View
  accessibilityLabel="button-icon"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  hitSlop={
    {
      "bottom": 15,
      "left": 15,
      "right": 15,
      "top": 15,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="button-icon"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 16,
        },
        {
          "color": "black",
        },
        {
          "fontFamily": "Icon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    Ó§Å
  </Text>
</View>
`;

exports[`Icon component test Icon without Name snapshot test should render correctly 1`] = `null`;
